traffic_ops/client/stats_summary.go

```diff
@@ -15,111 +15,69 @@ package client
 import (
 	"encoding/json"
 	"fmt"
-	"strconv"
+	"net"
+	"net/url"
 
 	tc "github.com/apache/trafficcontrol/lib/go-tc"
 )
 
-// SummaryStats ...
-// Deprecated: use GetSummaryStats
-func (to *Session) SummaryStats(cdn string, deliveryService string, statName string) ([]tc.StatsSummary, error) {
-	ss, _, err := to.GetSummaryStats(cdn, deliveryService, statName)
-	return ss, err
-}
+const (
+	basePathSummaryStats = apiBase + "/stats_summary"
+)
 
-func (to *Session) GetSummaryStats(cdn string, deliveryService string, statName string) ([]tc.StatsSummary, ReqInf, error) {
-	var queryParams []string
-	if len(cdn) > 0 {
-		queryParams = append(queryParams, fmt.Sprintf("cdnName=%s", cdn))
-	}
-	if len(deliveryService) > 0 {
-		queryParams = append(queryParams, fmt.Sprintf("deliveryServiceName=%s", deliveryService))
+// GetSummaryStats gets a list of summary stats with the ability to filter on cdn,deliveryService and/or stat
+func (to *Session) GetSummaryStats(cdn, deliveryService, statName *string) (tc.StatsSummaryResponse, ReqInf, error) {
```

Comment:
I don't know whether we're allowed to change the signature of client functions without a deprecation period. Removing that one above that was deprecated in 3.x should be fine, but I'm not sure about this one. Not that I don't like the improvements.

My knee-jerk reaction is to ask the mailing list about this.